pkg/zinit/commands.go

```diff
@@ -304,3 +323,29 @@ func (c *Client) Forget(service string) error {
 func (c *Client) Kill(service string, sig Signal) error {
 	return c.cmd(fmt.Sprintf("kill %s %s", service, string(sig)), nil)
 }
+
+// Destroy a service completely (stop, forget and remove) if all filters are matched
+func (c *Client) Destroy(service string, filters ...Filter) error {
+	initService, err := c.Get(service)
+	if err != nil {
+		// service is not there, mostly
+		return nil
+	}
+
+	// FIXME: should we allow empty filter so it would only match with service name?
+	for _, filter := range filters {
+		if !filter.matches(&initService) {
+			return nil
+		}
+	}
+
+	if err := c.StopWait(30*time.Second, service); err != nil {
+		return err
+	}
```

Comment:
Because you can actually destroy multiple services (check the comments above) it means you need to use the StopAll method instead

Again, we are trying to make this as generic as possible and not crafted for a specific use case